core/frontend/src/types/autopilot.ts

```diff
@@ -0,0 +1,62 @@
+export interface Firmware {
+    name: string
+    url: URL
+}
+
+export enum Vehicle {
+    Sub = 'Sub',
+    Rover = 'Rover',
+    Plane = 'Plane',
+    Copter = 'Copter',
+}
+
+export enum SITLFrame {
```

Comment:
Is it possible somehow to get these enums from the backend ? To avoid maintaining two enums in sync.

Reply:
Maybe...
I like the idea, but have to think on how to do it properly, since languages are different and so are restrictions (last time I've checked the enum support on json is not yet rounded for both languages).

I've opened issue #451 so we can study that for the other types as well.

I suspect we will have problems with type-check and auto-complete, though, since they would be dynamically loaded.